File: 2000 98 & ME/QuickStart/howto/samples/Services/PerformanceCounters/pcwrite/pcwrite.cs
using System;
using System.Diagnostics;
using System.Threading;

public class PCWrite{

	public static void Main(string[] args){

		string appName = Environment.GetCommandLineArgs()[0];

		if(args.Length != 3)
		{
			Console.WriteLine("Usage: " + appName + " <object> <counter> <instance>");
			return;
		}

		string objectName	= args[0];
		string counterName	= args[1];
		string instanceName = args[2];

		if(!PerformanceCounter.CategoryExists(objectName))
		{
			PerformanceCounterCategory.Create(objectName,"Simple Counter Object",counterName,"Simple Counter");
		}

		Console.WriteLine("Category Exists");

		Thread.Sleep(30000);
		if(!PerformanceCounter.Exists(objectName,counterName))
		{
			Console.WriteLine("The counter does not exists!");
			return;
		}

		Console.WriteLine("Counter Exists");

		PerformanceCounter aCounter = new PerformanceCounter(objectName, counterName ,instanceName);
		aCounter.RawValue = 50;

		Console.WriteLine("Press \'q\' to quit the sample");   
		Console.WriteLine("Press \'+\' to increment the counter");
		Console.WriteLine("Press \'-\' to decrement the counter");

		Console.WriteLine("Started");
		int command;
		do
		{
			command = Console.Read();
			if(command=='-') aCounter.IncrementBy(-5);
			if(command=='+') aCounter.IncrementBy(5);

			Thread.Sleep(500);
		}
		while(command!='q');
	}
}

File: 2000 98 & ME/QuickStart/howto/samples/Services/PerformanceCounters/pcwrite/vb/pcwrite.vb
Imports System
Imports System.Diagnostics
Imports System.Threading

public class PCWrite
	public shared sub Main()
		Dim appName As string = Environment.GetCommandLineArgs()(0)

		if(Environment.GetCommandLineArgs().Length <> 4) then
			Console.WriteLine("Usage: " + appName + " <object> <counter> <instance>")
			exit sub
		end if

		Dim objectName As String = Environment.GetCommandLineArgs()(1)
		Dim counterName As String = Environment.GetCommandLineArgs()(2)
		Dim instanceName As String = Environment.GetCommandLineArgs()(3)

		Dim trim(1) As Char
		trim(0) = " "c
		objectName = objectName.TrimEnd(trim)
		counterName = counterName.TrimEnd(trim)
		instanceName = instanceName.TrimEnd(trim)

		if(not PerformanceCounter.CategoryExists(objectName)) then
			PerformanceCounterCategory.Create(objectName,"Simple Counter Object",counterName,"Simple Counter")
		end if

		if(not PerformanceCounter.Exists(objectName,counterName)) then
			Console.WriteLine("The counter does not exists!")
			exit sub
		end if

		Dim aCounter As PerformanceCounter = new PerformanceCounter(objectName, counterName ,instanceName)
		aCounter.RawValue = 50

		Console.WriteLine("Press 'q' to quit the sample")	
		Console.WriteLine("Press '+' to increment the counter")
		Console.WriteLine("Press '-' to decrement the counter")

		Console.WriteLine("Started")
		Dim command as Integer
		Do
			command = Console.Read()
			if(command=45) then 
				aCounter.IncrementBy(-5)
			end if
			if(command=43) then 
				aCounter.IncrementBy(5)
			end if
		loop until command=113
	end sub
end class
